File: frontend/src/types/player.ts
export type PlayerRole = 'batsman' | 'bowler' | 'all_rounder' | 'wicket_keeper'

export interface Player {
  id: number
  team_id?: number
  name: string
  role: PlayerRole
  batting_style?: string
  bowling_style?: string
  country: string
  age: number
  birth_date?: string
  auction_price?: number
  image_url?: string
  has_projection?: boolean
}

export interface MatchPerformance {
  match_id: number
  date: string
  opponent: string
  runs: number
  balls_faced: number
  wickets: number
}

export interface CareerStats {
  matches_played: number
  runs_scored: number
  batting_average: number
  strike_rate: number
  highest_score: number
  fours: number
  sixes: number
  fifties: number
  hundreds: number
  wickets_taken: number
  bowling_average?: number
  economy_rate?: number
  best_bowling_figures?: string
  five_wickets: number
}

export interface RecentForm {
  last_5_matches: MatchPerformance[]
  trend: 'improving' | 'declining' | 'stable'
}

export interface SeasonBattingStats {
  matches: number
  runs: number
  highest_score: number
  average: number
  strike_rate: number
  balls_faced: number
  fours: number
  sixes: number
}

export interface SeasonBowlingStats {
  matches: number
  balls: number
  runs: number
  wickets: number
  average: number
  economy: number
  strike_rate: number
  best_figures?: string
  five_wickets: number
}

export interface SeasonStats {
  season: number
  team: string
  batting: SeasonBattingStats
  bowling: SeasonBowlingStats
}

export interface PlayerDetail extends Player {
  international_caps: number
  career_stats: CareerStats
  season_stats: SeasonStats[]
  recent_form: RecentForm
}

export interface PlayerProjection {
  player_id: number
  team_id?: number
  player_name: string
  predicted_runs: number
  predicted_wickets: number
  features: Record<string, number>
}

export interface PlayerStats {
  player_id: number
  season?: number
  batting: Record<string, number>
  bowling?: Record<string, number>
}

File: backend/app/schemas/player.py
"""Player schema definitions."""
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class Player(BaseModel):
    id: int
    team_id: Optional[int] = None
    name: str
    role: str
    batting_style: Optional[str] = None
    bowling_style: Optional[str] = None
    country: str
    age: int
    birth_date: Optional[str] = None
    auction_price: Optional[float] = None
    image_url: Optional[str] = None
    has_projection: Optional[bool] = None

    class Config:
        orm_mode = True


class MatchPerformance(BaseModel):
    match_id: int
    date: str
    opponent: str
    runs: int
    balls_faced: int
    wickets: int


class CareerStats(BaseModel):
    matches_played: int
    runs_scored: int
    batting_average: float
    strike_rate: float
    highest_score: int
    fours: int
    sixes: int
    fifties: int
    hundreds: int
    wickets_taken: int
    bowling_average: Optional[float] = None
    economy_rate: Optional[float] = None
    best_bowling_figures: Optional[str] = None
    five_wickets: int


class RecentForm(BaseModel):
    last_5_matches: List[MatchPerformance]
    trend: str


class SeasonBattingStats(BaseModel):
    matches: int
    runs: int
    highest_score: int
    average: float
    strike_rate: float
    balls_faced: int
    fours: int
    sixes: int


class SeasonBowlingStats(BaseModel):
    matches: int
    balls: int
    runs: int
    wickets: int
    average: float
    economy: float
    strike_rate: float
    best_figures: Optional[str] = None
    five_wickets: int


class SeasonStats(BaseModel):
    season: int
    team: str
    batting: SeasonBattingStats
    bowling: SeasonBowlingStats


class PlayerDetail(Player):
    international_caps: int
    career_stats: CareerStats
    season_stats: List[SeasonStats]
    recent_form: RecentForm


class PlayerStats(BaseModel):
    player_id: int
    season: Optional[int] = None
    batting: dict
    bowling: Optional[dict]
